### ISSUE TITLE
Dismiss engage and disengage survey and not reset the delay clock

### ISSUE THREAD

**Issue by YiranWangPARC14** [secure-software-engineering/rose#218]
I have observed both me and a pilot, that when we dismissed a disengage survey because we were not leaving FB yet, we oftentimes left not long after we dismissed the disengage survey (e.g., 1 minute later). For these instances, since the disengage survey's delay clock is reset to the next one (1-hour later, or 30-min later per the last ticket), when we actually leave FB, the disengage survey is not showing up.

So I was wondering, if there is a way to not reset the delay clock when the user dismiss the survey. This is for both engage and disengage survey. For example:

If the last time a disengage survey is triggered at 10am, and between 10:55am - 11:02am I am using Facebook. At 11am, the disengage survey is triggered because I was on another site between 10:58-11am, but I was planning to go back to FB, so I click "I'm not leaving FB yet." and continue to use FB until 11:02am. At 11:02am, I log out of FB. Is there a way for the disengage survey to pop up again at 11:02am?

In other words, if the the user doesn't dismiss the survey, the trigger follow a regular time interval (be it 1hour or 30 minute).

If the user dismiss the survey, the timer continues to record the time lapse since last survey trigger instead of reset to 0, so that the next time the user leaves FB, the survey can pop up again.

**Comment by dharma-guardian** [comment]
OK, I anticipated you might desire this behavior.
Right now the implementation is: How often should a survey pop up?
And you want it changed to: How often should participants complete a survey?
Right?

The tracking works a little different. It is not actually a timer/clock, which I reset, but when a engage/disengage is observed it checks: when was the last time x was triggered. It is a possible, but there might be a catch, because I need a new variable. Otherwise you would get asked every few seconds, wether you left facebook or not until you complete the survey. I'll further look into it tomorrow.

**Comment by YiranWangPARC14** [comment]
Right. I want the participants to complete a survey (pre- or post- FB) if 1) they meet the criteria of engaging or leaving FB now and 2) the last one they complete was more than 30 minutes ago.

Thanks for explaining.

**Comment by dharma-guardian** [comment]
This needs testing, because of significant changes.
